shaded jar shades too much

To make `@Grab` annotations and some Rhino report work, I need to exclude 3 more things from shading:
* org.apache.ivy
* org.mozilla
* org.xml.sax

fresh 1.17-SNAPSHOT build with shading exclusions:
```
                                    <excludes>
                                        <exclude>org.codehaus.groovy.**</exclude>
                                        <exclude>org.w3c.**</exclude>
                                        <exclude>org.apache.ivy.**</exclude>
                                    </excludes>
``` 
:
```
ICP-AWS ServiceOp@iqos-non-prod tests$ java -version
java version "1.8.0_202"
Java(TM) SE Runtime Environment (build 1.8.0_202-b08)
Java HotSpot(TM) 64-Bit Server VM (build 25.202-b08, mixed mode)
ICP-AWS ServiceOp@iqos-non-prod tests$ java -cp "/Users/clemens/tools/webtau/1.17-SNAPSHOT/dist/lib/*" -Dgrape.config=/Users/clemens/ws/tsiq-api/tests/resources/grapeConfig.xml com.twosigma.webtau.cli.WebTauCliApp --workingDir=src/main/groovy scenarios
Exception in thread "main" java.lang.NoClassDefFoundError: webtaushaded/org/xml/sax/helpers/DefaultHandler
        at java.lang.ClassLoader.defineClass1(Native Method)
        at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
        at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
        at java.net.URLClassLoader.defineClass(URLClassLoader.java:468)
        at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        at org.apache.ivy.core.settings.IvySettings.getSettingsURL(IvySettings.java:475)
        at org.apache.ivy.core.settings.IvySettings.getDefaultSettingsURL(IvySettings.java:461)
        at org.apache.ivy.core.settings.IvySettings.getDefaultSettingsDir(IvySettings.java:469)
        at org.apache.ivy.core.settings.IvySettings.<init>(IvySettings.java:227)
        at org.apache.ivy.core.settings.IvySettings.<init>(IvySettings.java:222)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:80)
        at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrapNoCoerce.callConstructor(ConstructorSite.java:105)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallConstructor(CallSiteArray.java:59)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:237)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:241)
        at groovy.grape.GrapeIvy.<init>(GrapeIvy.groovy:100)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at java.lang.Class.newInstance(Class.java:442)
        at groovy.grape.Grape.getInstance(Grape.java:123)
        at groovy.grape.Grape.grab(Grape.java:157)
        at groovy.grape.GrabAnnotationTransformation.visit(GrabAnnotationTransformation.java:376)
        at org.codehaus.groovy.transform.ASTTransformationVisitor$3.call(ASTTransformationVisitor.java:318)
        at org.codehaus.groovy.control.CompilationUnit.applyToSourceUnits(CompilationUnit.java:965)
        at org.codehaus.groovy.control.CompilationUnit.doPhaseOperation(CompilationUnit.java:647)
        at org.codehaus.groovy.control.CompilationUnit.processPhaseOperations(CompilationUnit.java:623)
        at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:600)
        at groovy.lang.GroovyClassLoader.doParseClass(GroovyClassLoader.java:390)
        at groovy.lang.GroovyClassLoader.access$300(GroovyClassLoader.java:89)
        at groovy.lang.GroovyClassLoader$5.provide(GroovyClassLoader.java:330)
        at groovy.lang.GroovyClassLoader$5.provide(GroovyClassLoader.java:327)
        at org.codehaus.groovy.runtime.memoize.ConcurrentCommonCache.getAndPut(ConcurrentCommonCache.java:147)
        at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:325)
        at groovy.util.GroovyScriptEngine$ScriptClassLoader.doParseClass(GroovyScriptEngine.java:242)
        at groovy.util.GroovyScriptEngine$ScriptClassLoader.parseClass(GroovyScriptEngine.java:229)
        at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:309)
        at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:251)
        at groovy.util.GroovyScriptEngine.loadScriptByName(GroovyScriptEngine.java:545)
        at groovy.util.GroovyScriptEngine.createScript(GroovyScriptEngine.java:595)
        at groovy.util.GroovyScriptEngine$createScript.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:115)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:135)
        at com.twosigma.webtau.cfg.WebTauGroovyFileConfigHandler.onAfterCreate(WebTauGroovyFileConfigHandler.groovy:48)
        at com.twosigma.webtau.cfg.WebTauConfig.lambda$triggerConfigHandlers$16(WebTauConfig.java:117)
        at java.util.ArrayList.forEach(ArrayList.java:1257)
        at com.twosigma.webtau.cfg.WebTauConfig.triggerConfigHandlers(WebTauConfig.java:117)
        at com.twosigma.webtau.cfg.WebTauConfig.<init>(WebTauConfig.java:105)
        at com.twosigma.webtau.cfg.WebTauConfig$CfgInstanceHolder.<clinit>(WebTauConfig.java:377)
        at com.twosigma.webtau.cfg.WebTauConfig.getCfg(WebTauConfig.java:78)
        at com.twosigma.webtau.cfg.WebTauConfig$getCfg$1.callStatic(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallStatic(CallSiteArray.java:55)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:196)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:200)
        at com.twosigma.webtau.cli.WebTauCliApp.main(WebTauCliApp.groovy:67)
Caused by: java.lang.ClassNotFoundException: webtaushaded.org.xml.sax.helpers.DefaultHandler
        at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        ... 68 more
```

for both java 8 and 11, I get this mozilla exception:
```
Errored tests:
[x] compression POST response scenarios/compression.groovy
com.twosigma.webtau.http.HttpException: error during http.post(https://api-integ.iqos-np.twosigmaiq.com/products/golden/schema)
        at scenarios.compression$_run_closure2.doCall(compression.groovy:33)
        at scenarios.compression$_run_closure2.doCall(compression.groovy)
Caused by: webtaushaded.com.google.common.util.concurrent.ExecutionError: webtaushaded.com.google.common.util.concurrent.ExecutionError: java.lang.NoClassDefFoundError: org/mozilla/javascript/Context
        at webtaushaded.com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2216)
        at webtaushaded.com.google.common.cache.LocalCache.get(LocalCache.java:4147)
        at webtaushaded.com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:4151)
        at webtaushaded.com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:5140)
        at webtaushaded.com.github.fge.jsonschema.core.processing.CachingProcessor.process(CachingProcessor.java:122)
        at webtaushaded.com.github.fge.jsonschema.processors.validation.InstanceValidator.process(InstanceValidator.java:109)
        at webtaushaded.com.github.fge.jsonschema.processors.validation.ValidationProcessor.process(ValidationProcessor.java:56)
        at webtaushaded.com.github.fge.jsonschema.processors.validation.ValidationProcessor.process(ValidationProcessor.java:34)
        at webtaushaded.com.github.fge.jsonschema.core.processing.ProcessingResult.of(ProcessingResult.java:79)
        at webtaushaded.com.github.fge.jsonschema.main.JsonSchemaImpl.doValidate(JsonSchemaImpl.java:77)
        at webtaushaded.com.github.fge.jsonschema.main.JsonSchemaImpl.validate(JsonSchemaImpl.java:100)
        at webtaushaded.com.atlassian.oai.validator.schema.SchemaValidator.doValidate(SchemaValidator.java:137)
        at webtaushaded.com.atlassian.oai.validator.schema.SchemaValidator.validate(SchemaValidator.java:112)
        at webtaushaded.com.atlassian.oai.validator.interaction.RequestValidator.validateBody(RequestValidator.java:366)
        at webtaushaded.com.atlassian.oai.validator.interaction.RequestValidator.validateRequestBody(RequestValidator.java:295)
        at webtaushaded.com.atlassian.oai.validator.interaction.RequestValidator.validateRequest(RequestValidator.java:100)
        at webtaushaded.com.atlassian.oai.validator.SwaggerRequestResponseValidator.lambda$validate$0(SwaggerRequestResponseValidator.java:140)
        at webtaushaded.com.atlassian.oai.validator.SwaggerRequestResponseValidator.validateOnApiOperation(SwaggerRequestResponseValidator.java:214)
        at webtaushaded.com.atlassian.oai.validator.SwaggerRequestResponseValidator.validate(SwaggerRequestResponseValidator.java:137)
Caused by: webtaushaded.com.google.common.util.concurrent.ExecutionError: java.lang.NoClassDefFoundError: org/mozilla/javascript/Context
        at webtaushaded.com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2216)
        at webtaushaded.com.google.common.cache.LocalCache.get(LocalCache.java:4147)
        at webtaushaded.com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:4151)
        at webtaushaded.com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:5140)
        at webtaushaded.com.github.fge.jsonschema.core.processing.CachingProcessor.process(CachingProcessor.java:122)
        at webtaushaded.com.github.fge.jsonschema.processors.validation.ValidationChain.process(ValidationChain.java:107)
        at webtaushaded.com.github.fge.jsonschema.processors.validation.ValidationChain.process(ValidationChain.java:57)
        at webtaushaded.com.github.fge.jsonschema.core.processing.ProcessorMap$Mapper.process(ProcessorMap.java:166)
        at webtaushaded.com.github.fge.jsonschema.core.processing.ProcessingResult.of(ProcessingResult.java:79)
        at webtaushaded.com.github.fge.jsonschema.core.processing.CachingProcessor$1.load(CachingProcessor.java:141)
        at webtaushaded.com.github.fge.jsonschema.core.processing.CachingProcessor$1.load(CachingProcessor.java:133)
        at webtaushaded.com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3708)
        at webtaushaded.com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2416)
        at webtaushaded.com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2299)
        at webtaushaded.com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2212)
Caused by: java.lang.NoClassDefFoundError: org/mozilla/javascript/Context
        at webtaushaded.org.mozilla.javascript.optimizer.Codegen.createScriptObject(Codegen.java:89)
        at webtaushaded.org.mozilla.javascript.Context.compileImpl(Context.java:2394)
        at webtaushaded.org.mozilla.javascript.Context.compileString(Context.java:1335)
        at webtaushaded.org.mozilla.javascript.Context.compileString(Context.java:1324)
        at webtaushaded.org.mozilla.javascript.Context.evaluateString(Context.java:1076)
        at webtaushaded.com.github.fge.jsonschema.core.util.RhinoHelper.<clinit>(RhinoHelper.java:94)
        at webtaushaded.com.github.fge.jsonschema.core.keyword.syntax.checkers.common.PatternSyntaxChecker.checkValue(PatternSyntaxChecker.java:62)
        at webtaushaded.com.github.fge.jsonschema.core.keyword.syntax.checkers.AbstractSyntaxChecker.checkSyntax(AbstractSyntaxChecker.java:116)
        at webtaushaded.com.github.fge.jsonschema.core.keyword.syntax.SyntaxProcessor.validate(SyntaxProcessor.java:112)
        at webtaushaded.com.github.fge.jsonschema.core.keyword.syntax.SyntaxProcessor.validate(SyntaxProcessor.java:118)
        at webtaushaded.com.github.fge.jsonschema.core.keyword.syntax.SyntaxProcessor.validate(SyntaxProcessor.java:118)
        at webtaushaded.com.github.fge.jsonschema.core.keyword.syntax.SyntaxProcessor.rawProcess(SyntaxProcessor.java:70)
        at webtaushaded.com.github.fge.jsonschema.core.keyword.syntax.SyntaxProcessor.rawProcess(SyntaxProcessor.java:45)
        at webtaushaded.com.github.fge.jsonschema.core.processing.RawProcessor.process(RawProcessor.java:77)
        at webtaushaded.com.github.fge.jsonschema.core.processing.RawProcessor.process(RawProcessor.java:41)
        at webtaushaded.com.github.fge.jsonschema.core.processing.ProcessorChain$ProcessorMerger.process(ProcessorChain.java:190)
        at webtaushaded.com.github.fge.jsonschema.core.processing.ProcessingResult.of(ProcessingResult.java:79)
        at webtaushaded.com.github.fge.jsonschema.core.processing.CachingProcessor$1.load(CachingProcessor.java:141)
        at webtaushaded.com.github.fge.jsonschema.core.processing.CachingProcessor$1.load(CachingProcessor.java:133)
        at webtaushaded.com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3708)
        at webtaushaded.com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2416)
        at webtaushaded.com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2299)
        at webtaushaded.com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2212)
Caused by: java.lang.ClassNotFoundException: org.mozilla.javascript.Context
        at webtaushaded.org.mozilla.javascript.DefiningClassLoader.loadClass(DefiningClassLoader.java:42)

```

--- a/webtau-report/src/main/java/com/twosigma/webtau/report/HtmlReportGenerator.java
+++ b/webtau-report/src/main/java/com/twosigma/webtau/report/HtmlReportGenerator.java
@@ -16,6 +16,9 @@
 
 package com.twosigma.webtau.report;
 
+import static com.twosigma.webtau.cfg.WebTauConfig.getCfg;
+import static java.util.stream.Collectors.toList;
+
 import com.twosigma.webtau.cfg.ConfigValue;
 import com.twosigma.webtau.cfg.WebTauMeta;
 import com.twosigma.webtau.console.ConsoleOutputs;
@@ -25,7 +28,6 @@ import com.twosigma.webtau.reporter.WebTauTest;
 import com.twosigma.webtau.utils.FileUtils;
 import com.twosigma.webtau.utils.JsonUtils;
 import com.twosigma.webtau.utils.ResourceUtils;
-
 import java.nio.file.Path;
 import java.util.Base64;
 import java.util.LinkedHashMap;
@@ -33,9 +35,6 @@ import java.util.List;
 import java.util.Map;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
-
-import static com.twosigma.webtau.cfg.WebTauConfig.getCfg;
-import static java.util.stream.Collectors.toList;
 
 public class HtmlReportGenerator implements ReportGenerator {
     private String css;
@@ -132,3 +131,4 @@ public class HtmlReportGenerator implements ReportGenerator {
         return result;
     }
 }
+

--- a/webtau-report/src/main/java/com/twosigma/webtau/report/HtmlReportGenerator.java
+++ b/webtau-report/src/main/java/com/twosigma/webtau/report/HtmlReportGenerator.java
@@ -131,4 +131,3 @@ public class HtmlReportGenerator implements ReportGenerator {
         return result;
     }
 }
-

--- a/webtau-shaded/pom.xml
+++ b/webtau-shaded/pom.xml
@@ -64,6 +64,9 @@
                                     <excludes>
                                         <exclude>org.codehaus.groovy.**</exclude>
                                         <exclude>org.w3c.**</exclude>
+                                        <exclude>org.apache.ivy.**</exclude>
+                                        <exclude>org.mozilla.**</exclude>
+                                        <exclude>org.xml.sax.**</exclude>
                                     </excludes>
                                 </relocation>
                                 <relocation>

